### PR TITLE
Update Reminder* functions for intent-triggering

### PR DESCRIPTION
**Pull request by JEM-Mosig**
**Proposed changes**:
- Replace the `action` with an `intent` parameter for `ReminderScheduled` and add an `entity` parameter
- Replace the `action` with an `intent` parameter for `ReminderCancelled`

The release should be synced with the release of https://github.com/RasaHQ/rasa/pull/4964 in Rasa.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

File: rasa_sdk/events.py [hunk 1]
```diff
@@ -68,16 +68,18 @@ def AllSlotsReset(timestamp: Optional[float] = None) -> EventType:
 
 # noinspection PyPep8Naming
 def ReminderScheduled(
-    action_name: Text,
+    intent: Text,
     trigger_date_time: datetime.datetime,
+    entities: Union[List[Dict[Text, Any]], Dict[Text, Text]] = None,
```
Comment (wochinge) [inline rasa_sdk/events.py]:
```suggestion
    entities: Optional[Union[List[Dict[Text, Any]], Dict[Text, Text]]] = None,
```

File: rasa_sdk/events.py [hunk 2]
```diff
@@ -86,12 +88,16 @@ def ReminderScheduled(
 
 # noinspection PyPep8Naming
 def ReminderCancelled(
-    action_name: Text, name: Optional[Text] = None, timestamp: Optional[float] = None
+    name: Optional[Text] = None,
+    intent_name: Optional[Text] = None,
+    entities: Union[List[Dict[Text, Any]], Dict[Text, Text]] = None,
```
Comment (wochinge) [inline rasa_sdk/events.py]:
```suggestion
    entities: Optional[Union[List[Dict[Text, Any]], Dict[Text, Text]]] = None,
```

**Comment by JEM-Mosig** [comment]
> Also is there a way that we can make this backwards compatible?
> E.g. by keeping the `action` parameter and returning the event in the old way in case `action` was given?

I don't think that's a good idea. If users want to use the old version of reminders, then they have to use the old version of Rasa anyway, and that should always be combined with the old version of the Rasa SDK. Don't you think?

File: rasa_sdk/events.py [hunk 3]
```diff
@@ -86,12 +88,16 @@ def ReminderScheduled(
 
 # noinspection PyPep8Naming
 def ReminderCancelled(
-    action_name: Text, name: Optional[Text] = None, timestamp: Optional[float] = None
+    name: Optional[Text] = None,
+    intent_name: Optional[Text] = None,
+    entities: Optional[Union[List[Dict[Text, Any]], Dict[Text, Text]]] = None,
+    timestamp: Optional[float] = None,
 ) -> EventType:
```
Comment (wochinge) [inline rasa_sdk/events.py]: can we add a warning in case `intent_name` starts with `utter_` or `action_`? That's also sth we could unit test then.

Comment (JEM-Mosig) [reply rasa_sdk/events.py]: Done. I am not sure if it was correct to declare it as `DeprecationWarning`, though.

Comment (JEM-Mosig) [reply rasa_sdk/events.py]: Oh, and doing unit tests now...

Comment (wochinge) [reply rasa_sdk/events.py]: Should be a `FutureWarning`

Comment (JEM-Mosig) [reply rasa_sdk/events.py]: Ah, ok. I thought it's meant for the developers who develop the bot. Not for the enduser. So why is it `Future`?

Comment (JEM-Mosig) [reply rasa_sdk/events.py]: I've changed it now.

**Comment by wochinge** [comment]
Mhm, actually doing a review of your own code can be quite useful.

File: rasa_sdk/events.py [hunk 4]
```diff
@@ -68,16 +69,30 @@ def AllSlotsReset(timestamp: Optional[float] = None) -> EventType:
...
 def ReminderScheduled(
-    action_name: Text,
+    intent_name: Text,
     trigger_date_time: datetime.datetime,
+    entities: Optional[Union[List[Dict[Text, Any]], Dict[Text, Text]]] = None,
     name: Optional[Text] = None,
     kill_on_user_message: bool = True,
     timestamp: Optional[float] = None,
 ) -> EventType:
+    if intent_name.startswith("utter_"):
```
Comment (wochinge) [inline rasa_sdk/events.py]: optional: One warning for both cases is fine imo, but up to you.

File: tests/test_events.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import pytest
+
+import warnings
+from datetime import datetime
+
+import rasa_sdk.events as events
+
+
+def test_reminder_scheduled_correctly():
+    with warnings.catch_warnings(record=True) as warning:
```
Comment (wochinge) [inline tests/test_events.py]: can we please test the warnings as described here https://docs.pytest.org/en/latest/warnings.html#warns (`with pytest.warns(FutureWarning) ...`)

File: tests/test_events.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        events.ReminderScheduled("greet", datetime.now())
+        assert len(warning) == 0
+
+
+def test_reminder_scheduled_with_action():
```
Comment (wochinge) [inline tests/test_events.py]: and please use `pytest.parametrize` for the tests, makes them a lot shorter :-)

**Comment by JEM-Mosig** [comment]
> Mhm, actually doing a review of your own code can be quite useful.

Ah. I thought this was a malfunction, or I clicked something wrong, and I was confused because there was no way to 'cancel'. So I can compare my code to master there... yes, that's actually useful. Sorry, still getting used to GitHub.

File: rasa_sdk/events.py [hunk 7]
```diff
@@ -86,12 +97,24 @@ def ReminderScheduled(
 
 # noinspection PyPep8Naming
 def ReminderCancelled(
-    action_name: Text, name: Optional[Text] = None, timestamp: Optional[float] = None
+    name: Optional[Text] = None,
+    intent_name: Optional[Text] = None,
+    entities: Optional[Union[List[Dict[Text, Any]], Dict[Text, Text]]] = None,
+    timestamp: Optional[float] = None,
 ) -> EventType:
+    if intent_name and (
```
Comment (wochinge) [inline rasa_sdk/events.py]: that's duplicate. can we put it in a function and re-use it?

File: tests/test_events.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import pytest
+from datetime import datetime
+
+import rasa_sdk.events as events
+
+
```
Comment (wochinge) [inline tests/test_events.py]: 💯

File: rasa_sdk/events.py (hunk 7 above)
Comment (JEM-Mosig) [reply rasa_sdk/events.py]: Is it ok to call the function `_is_probably_action_name`? I know it's not an "intent" like `_check_if_probably_action_name`, but in this case the former is actually more clear.

Comment (wochinge) [reply rasa_sdk/events.py]: good to me :-)

Comment (wochinge) [reply rasa_sdk/events.py]: wait, `is_...` indicates that the function returns a `bool` to me :-D Maybe something like `_check_potential_action_usage`?

Comment (JEM-Mosig) [reply rasa_sdk/events.py]: It does return `bool`. I just pushed it, for clarification.

Comment (JEM-Mosig) [reply rasa_sdk/events.py]: The messages are different for `ReminderScheduled` and `ReminderCancelled`, so I only "outsourced" the condition.

**Pull request merged**